### PR TITLE
Fetch release-please manifest and config files from changes-branch

### PR DESCRIPTION
**Pull request by dgellow**
When using release-please with `--change-branch=next` against a new repository where no manifest or config has been merged to `main` yet, we get an error:

```
/home/sam/stainless/stainless/gh-app/node_modules/@stainless-api/release-please/src/manifest.ts:1596
      throw new ConfigurationError(
            ^
ConfigurationError: base (DefinitelyATestOrg/sdk-node): Missing required manifest config: release-please-config.json
    at fetchManifestConfig (/home/sam/stainless/stainless/gh-app/node_modules/@stainless-api/release-please/src/manifest.ts:1596:13)
    at async parseConfig (/home/sam/stainless/stainless/gh-app/node_modules/@stainless-api/release-please/src/manifest.ts:1541:18)
    at async Promise.all (index 0)
    at async Function.fromManifest (/home/sam/stainless/stainless/gh-app/node_modules/@stainless-api/release-please/src/manifest.ts:417:9)
    at async runReleasePlease (/home/sam/stainless/stainless/gh-app/src/helpers/run-release-please.ts:47:20) {
  releaserName: 'base',
  repository: 'DefinitelyATestOrg/sdk-node'
}
```

My [test repositor](https://github.com/DefinitelyATestOrg/sdk-node) has a `next` branch with a manifest, but no release has been created yet, so `main` is just an "Initial commit".

The way upstream release-please generally work is that a `release-please bootstrap` command is first run to generate a default manifest and config. But in the context of Stainless that step isn't necessary, content pushed to `next` will always have release-please config files, we just need to be sure `release-please` is looking at the correct branch. Given that `next` is always supposed to be based on top of `main` we can consider its manifest as up to date.

### CODE REVIEW

**Comment by dgellow** [comment]
That will benefit both Stainless CI (Robert faced this issue) and the GitHub App (that's where I personally faced the issue when running against my freshly created test repo).

**Comment by dgellow** [comment]
You can see the generated PR here: https://github.com/DefinitelyATestOrg/sdk-node/pull/1

File: package.json
Comment (RobertCraigie) [inline package.json:9]: question: why were these removed?

File: src/strategies/base.ts
Comment (dgellow) [inline src/strategies/base.ts:755]: That's more in line with initial version numbers Stainless uses.

**Review by RobertCraigie — APPROVED** [review]

File: src/strategies/go-yoshi.ts
Comment (dgellow) [inline src/strategies/go-yoshi.ts:201]: Not sure why the initial version number is different for different languages? That's kind of odd and doesn't match Stainless behaviour.

File: package.json
Comment (dgellow) [reply package.json:9]: Since https://github.com/stainless-api/release-please/pull/28 this fork of release-please requires node v18+.

**Comment by dgellow** [comment]
@RobertCraigie Unless that's blocking you I will wait for Marcel to be able to review before merging. Just in case he disagrees or has some feedback.

**Review by marcel-stainless — APPROVED** [review]

**Pull request merged**
